marconi-core/src/Marconi/Core/Experiment/Transformer/WithResume.hs

```diff
@@ -57,16 +62,18 @@ Lens.makeLenses 'WithResume
 -- | A smart constructor for the @WithResume@ transformer
 withResume
   :: (Applicative m)
-  => (Word -> indexer event -> m [Point event])
-  -- ^ A way to extracte the last points of an indexer
-  -> Word
-  -- ^ SecurityParam
+  => (Point event -> indexer event -> m [Point event])
+  -- ^ A way to extract the last points of an indexer.
+  -- The points need to be sorted in ascending order.
+  -- TODO Why a function here? We would just provide [Point event]
```

Comment:
The idea was to save one step for the user and avoid mistake when we pass the events. It can be reconsidered though.

Reply:
When we pass the event? We don't pass any event IIUC.

What I meant by the TODO is that we can change the current API

```
        let getResumablePoints securityParam ix = do
              tes <- fmap (List.sortOn $ view Core.point) $ getIndexerEvents ix
              pure $ fmap (view Core.point) $ reverse $ take (fromIntegral securityParam + 1) $ reverse tes
        initialIndexerWithResume <-
          GenM.run $
            Core.withResume
              getResumablePoints
              k
              indexer
```

to 

```
        let k = ...
             indexer = ...
             resumablePoints = do
               tes <- fmap (List.sortOn $ view Core.point) $ getIndexerEvents indexer
               pure $ fmap (view Core.point) $ reverse $ take (fromIntegral k + 1) $ reverse tes
        initialIndexerWithResume <-
          GenM.run $
            Core.withResume
              resumablePoints
              indexer
```

I don't see the advantage of the current way.

Reply:
I meant "the points" and wrote "the events" sorry. The only advantage is that you avoid introducing the `resumablePoints` variable. I don't have a clear preference though, so I'd be happy with the change if you want to do it (and it may be useful if we want to pass arbitrary points).